src/msix/PAL/FileSystem/POSIX/DirectoryObject.cpp

```diff
@@ -60,15 +60,14 @@ namespace MSIX {
     {
         char* p = &path[startPos];
         if (*p == '/') { p++; }
-        while (*p != '\0')
+        for (; *p != '\0'; p++)

```

Comment:
>for (; *p != '\0'; p++) [](start = 8, length = 23)

Sorry that I missed this before, but this loop will now keep going past the null char.  Previously the termination would be that the inner while loop would reach the null, then create the last directory.  Afterward, the null char would prevent the iteration.

Now, the null char will be reached but the for iteration will be executed every time and push the pointer out past the end of the string.  We will keep mkdir'ing the same existing directory over and over until we reach two nulls in a row to terminate (or SEGSEV).